core/tests/test_utils.py

```diff
@@ -846,10 +996,10 @@ class FailingFunction(FunctionWrapper):
     def __init__(self, f, exception, num_tries_before_success):
         """Create a new Failing function.
 
-        args:
-          - f: see FunctionWrapper.
-          - exception: the exception to be raised.
-          - num_tries_before_success: the number of times to raise an
+        Args:
+            f: func. see FunctionWrapper.
+            exception: str. the exception to be raised.
```

Comment:
Are you sure this is a str? What happens when you do type(self._exception)?

Reply:
It should be str.

Reply:
Its type is Exception as this is the method that can raise any exception inside this base Exception class.